### ISSUE TITLE
[Grid Editor] Shortcut keys operations on a multiple selected cells impact the first row's cell

### ISSUE THREAD

**Issue by ZDabrowski** [robotframework/RIDE#2796]
The Grid shortcut keys operations on a multiple cells' selection impacts also the first row's cell.
It can be observed when using CTRL+SHIFT+I, CTRL+SHIFT+D, but also CTRL+X/CTRL+V.

**Environment**
Python 3.10.12
wxPython 4.2.1
robotframework 6.0.2
robotframework-ride 2.1.dev45
Ubuntu 22.04.4 LTS

**Steps**
1. Select a minimum 2 cells - horizontal/vertical selection, for example (1,5:1,6).
![obraz](https://github.com/robotframework/RIDE/assets/32097744/0ceec008-77ef-4f7b-94fb-0db212a88637)
2. Press shortcut CTRL+SHIFT+I to insert cells. The cell (1,1) was also moved.
![obraz](https://github.com/robotframework/RIDE/assets/32097744/bf2185d7-f9f8-4623-a39c-504e07b8a47b)
3. Select cells (1,5:1,6) again and then press shortcut CTRL+SHIFT+D to delete those cells. The cell (1,1) was also removed.
![obraz](https://github.com/robotframework/RIDE/assets/32097744/72c7fc27-b810-4306-a472-00bb06339ed0)
4. Select cells (1,5:1,6) again and press CTRL+SHIF+D to delete cells. The cell (1,1) was also moved.
![obraz](https://github.com/robotframework/RIDE/assets/32097744/1f6f0b7c-d2c5-4ad4-a804-b5c00c3abcf1)

I suppose, that the other operations have the same root cause. 
Additionally, the cut/paste operation of multiple cells (CTRL+X/CTL-V), incorrectly copied a value from cell (1,1) instead of previously selected cells.
1. Select cells (1,4:2,5)
![obraz](https://github.com/robotframework/RIDE/assets/32097744/9a10bccf-4716-4753-96a6-cf2403951bcc)
2. Cut selection. Noticed that the "active cell indicator" moves to the (1,1) grid position.
![obraz](https://github.com/robotframework/RIDE/assets/32097744/a326f242-fe1f-4655-bc13-a7b89bb7f2d8)
3. Paste the code to 9th row. Instead of (1,4:2,5) cells, there is an unexpected value from the (1,1) cell.
![obraz](https://github.com/robotframework/RIDE/assets/32097744/7dd0bbb9-1017-460c-8191-6bd5a3460487)

**Additional info**
Sample of code:
```robotframework
*** Test Cases ***
Test Rows
    Comment    First Line
    Run Keyword And Ignore Error    1
    Run Keyword And Ignore Error    2
    Run Keyword And Ignore Error    3
    Run Keyword And Ignore Error    4
    Run Keyword And Ignore Error    5
    Run Keyword And Ignore Error    6
    Run Keyword And Ignore Error    7
```

**Comment by HelioGuilherme66** [comment]
@ZDabrowski The first problem is fixed, but not the Ctrl-X/Ctrl-V and also Move Rows actions (Alt-Up/Alt-Down).
Tested on:
**Environment**
Python 3.12.3 (Virtual Env)
wxPython 4.2.2a1.dev5626+a1184286
Robot Framework 7.0
RIDE v2.1dev47
Fedora 40 (KDE/Plasma)

**Comment by HelioGuilherme66** [comment]
Still missing CTRL-X action :(

**Comment by HelioGuilherme66** [comment]
Fixed in v2.1dev49.
